ADAL/src/cache/ADTokenCache.m

```diff
@@ -69,6 +69,45 @@ - (void)dealloc
     SAFE_ARC_SUPER_DEALLOC();
 }
 
+- (id)cacheObjectForKey:(id<NSCopying>)key
+{
+    id object = nil;
+    @synchronized(self)
```

Comment:
I'm right now in the middle of taking a crack at this with r/w locks. Like I said the usage pattern of the ADAL cache is extremely read heavy so I'd like to avoid mutexes.